src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/FindArrayStatistics.cpp

```diff
@@ -400,22 +482,42 @@ void FindStatistics(const DataArray<T>& source, const Int32Array* featureIds, co
     auto* summationArray = dynamic_cast<Float32Array*>(arrays[6]);
     auto* histArray = dynamic_cast<Float32Array*>(arrays[7]);
 
-    ParallelDataAlgorithm dataAlg;
-    dataAlg.setRange(0, numFeatures);
-    dataAlg.execute(FindArrayStatisticsByIndexImpl<T>(inputValues->FindLength, inputValues->FindMin, inputValues->FindMax, inputValues->FindMean, inputValues->FindStdDeviation,
-                                                      inputValues->FindSummation, inputValues->FindHistogram, inputValues->MinRange, inputValues->MaxRange, inputValues->UseFullRange,
-                                                      inputValues->NumBins, mask, featureIds, source, lengthArray, minArray, maxArray, meanArray, stdDevArray, summationArray, histArray));
+    // ThreadSafeMessenger messenger(messageHandler, " Computing Statistics by Feature/Ensemble....");
+    // messenger.setTotalElements(numFeatures);
+    // messenger.setProgressIncrement(numFeatures / 100);
+
+    //    ParallelDataAlgorithm dataAlg;
+    //    dataAlg.setRange(0, numFeatures);
+    //    dataAlg.setParallelizationEnabled(true);
+    //    dataAlg.setGrainSize(500);
+    //    dataAlg.execute(FindArrayStatisticsByIndexImpl<T>(inputValues->FindLength, inputValues->FindMin, inputValues->FindMax, inputValues->FindMean, inputValues->FindStdDeviation,
+    //                                                      inputValues->FindSummation, inputValues->FindHistogram, inputValues->MinRange, inputValues->MaxRange, inputValues->UseFullRange,
+    //                                                      inputValues->NumBins, mask, featureIds, source, lengthArray, minArray, maxArray, meanArray, stdDevArray, summationArray, histArray,
+    //                                                      filter));
+#ifdef COMPLEX_ENABLE_MULTICORE
+    tbb::simple_partitioner simplePartitioner;
+    size_t grainSize = 500;
+    tbb::blocked_range<size_t> tbbRange(0, numFeatures, grainSize);
+    tbb::parallel_for(tbbRange,
+                      FindArrayStatisticsByIndexImpl<T>(inputValues->FindLength, inputValues->FindMin, inputValues->FindMax, inputValues->FindMean, inputValues->FindStdDeviation,
+                                                        inputValues->FindSummation, inputValues->FindHistogram, inputValues->MinRange, inputValues->MaxRange, inputValues->UseFullRange,
+                                                        inputValues->NumBins, mask, featureIds, source, lengthArray, minArray, maxArray, meanArray, stdDevArray, summationArray, histArray, filter),
+                      simplePartitioner);
+#endif
```

Comment:
If we are only letting the use calculate this statistic values by index when multicore is enabled, perhaps we should we be sending back a warning or something if they select one of these statistics with this by index option and don't have mutlicore enabled?

Reply:
Good catch.. I bet I never went back and figured out the normal array statistics bit when the "by index" is disabled.

Reply:
I addressed it partially by combining some of the stats algorithms like (min and max, summation/mean/stdDev) which help cut down on the time to run those algorithms.